### PR TITLE
Add a way to set a custom folder to store artifacts. The Screenshot Dir var is to be deprecated for this.

### PR DESCRIPTION
**Pull request by tirodkar**

### CODE REVIEW

File: EarlGrey/Common/GREYConfiguration.h [hunk 1]
```diff
@@ -143,12 +143,22 @@ GREY_EXTERN NSString *const kGREYConfigKeyIncludeStatusBarWindow;
...
  *
- *  Accepted values: NSString containing valid absolute filepath
+ *  Accepted values: NSString containing valid absolute filepath writable by the test.
  *  Default value: Documents directory of the app under test
  */
 GREY_EXTERN NSString *const kGREYConfigKeyScreenshotDirLocation;
 
 /**
+ *  Configuration for setting a directory location where any test artifacts such as screenshots,
+ *  test logs etc. are stored. The user should ensure that the location provided is writable by
```
Comment (khandpur) [inline EarlGrey/Common/GREYConfiguration.h]: logs, etc.

File: EarlGrey/Common/GREYConfiguration.m [hunk 2]
```diff
@@ -59,7 +60,11 @@ - (instancetype)init {
                                                                NSUserDomainMask,
                                                                YES);
     GREYFatalAssertWithMessage(searchPaths.count > 0, @"Couldn't find a valid documents directory");
+    NSLog(@"kGREYConfigKeyScreenshotDirLocation is to be deprecated in the "
+          "1.10.0 release. Please use the kGREYConfigKeyArtifactsDirLocation"
```
Comment (khandpur) [inline EarlGrey/Common/GREYConfiguration.m]: @"1.10.0

and

@"config key instead"

File: EarlGrey/Common/GREYConfiguration.m [hunk 3]
```diff
@@ -89,6 +94,11 @@ + (instancetype)sharedInstance {
 - (void)setValue:(id)value forConfigKey:(NSString *)configKey {
   GREYThrowOnNilParameter(value);
 
+  if ([configKey isEqualToString:kGREYConfigKeyScreenshotDirLocation]) {
+    NSLog(@"kGREYConfigKeyScreenshotDirLocation is to be deprecated in the "
+          "1.10.0 release. Please use the kGREYConfigKeyArtifactsDirLocation"
```
Comment (khandpur) [inline EarlGrey/Common/GREYConfiguration.m]: @"1.10.0

and

@"config key instead"

File: EarlGrey/Common/GREYConfiguration.h [hunk 4]
```diff
@@ -143,12 +143,22 @@ GREY_EXTERN NSString *const kGREYConfigKeyIncludeStatusBarWindow;
...
+ *  the test.
+ *
+ *  Accepted values: NSString containing a valid absolute filepath that is writable by the test.
+ *  Default value: Nil
```
Comment (khandpur) [inline EarlGrey/Common/GREYConfiguration.h]: Nil -> nil

File: EarlGrey/Common/GREYConfiguration.h (hunk 1 above)
Comment (tirodkar) [inline EarlGrey/Common/GREYConfiguration.h]: Done.

File: EarlGrey/Common/GREYConfiguration.h (hunk 4 above)
Comment (tirodkar) [inline EarlGrey/Common/GREYConfiguration.h]: @c nil

File: EarlGrey/Common/GREYConfiguration.m (hunk 2 above)
Comment (tirodkar) [inline EarlGrey/Common/GREYConfiguration.m]: Done

File: EarlGrey/Common/GREYConfiguration.m (hunk 3 above)
Comment (tirodkar) [inline EarlGrey/Common/GREYConfiguration.m]: Done

**Pull request merged**